homeassistant/components/anthropic/conversation.py

```diff
@@ -0,0 +1,316 @@
+"""Conversation support for Anthropic."""
+
+from collections.abc import Callable
+import json
+from typing import Any, Literal, cast
+
+import anthropic
+from anthropic._types import NOT_GIVEN
+from anthropic.types import (
+    Message,
+    MessageParam,
+    TextBlock,
+    TextBlockParam,
+    ToolParam,
+    ToolResultBlockParam,
+    ToolUseBlock,
+    ToolUseBlockParam,
+)
+import voluptuous as vol
+from voluptuous_openapi import convert
+
+from homeassistant.components import assist_pipeline, conversation
+from homeassistant.components.conversation import trace
+from homeassistant.const import CONF_LLM_HASS_API, MATCH_ALL
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError, TemplateError
+from homeassistant.helpers import device_registry as dr, intent, llm, template
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util import ulid
+
+from . import AnthropicConfigEntry
+from .const import (
+    CONF_CHAT_MODEL,
+    CONF_MAX_TOKENS,
+    CONF_PROMPT,
+    CONF_TEMPERATURE,
+    DOMAIN,
+    LOGGER,
+    RECOMMENDED_CHAT_MODEL,
+    RECOMMENDED_MAX_TOKENS,
+    RECOMMENDED_TEMPERATURE,
+)
+
+# Max number of back and forth with the LLM to generate a response
+MAX_TOOL_ITERATIONS = 10
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: AnthropicConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up conversation entities."""
+    agent = AnthropicConversationEntity(config_entry)
+    async_add_entities([agent])
+
+
+def _format_tool(
+    tool: llm.Tool, custom_serializer: Callable[[Any], Any] | None
+) -> ToolParam:
+    """Format tool specification."""
+    return ToolParam(
+        name=tool.name,
+        description=tool.description or "",
+        input_schema=convert(tool.parameters, custom_serializer=custom_serializer),
+    )
+
+
+class AnthropicConversationEntity(
+    conversation.ConversationEntity, conversation.AbstractConversationAgent
+):
+    """Anthropic conversation agent."""
+
+    _attr_has_entity_name = True
+    _attr_name = None
+
+    def __init__(self, entry: AnthropicConfigEntry) -> None:
+        """Initialize the agent."""
+        self.entry = entry
+        self.history: dict[str, list[MessageParam]] = {}
+        self._attr_unique_id = entry.entry_id
+        self._attr_device_info = dr.DeviceInfo(
+            identifiers={(DOMAIN, entry.entry_id)},
```

Comment:
Don't we get anything interesting from the API to use as unique id?

Reply:
No, the API is very basic. Also, we may use several config entries for the same key (but different prompt, etc)